nvtabular/ops/target_encoding.py

```diff
@@ -156,33 +139,21 @@ def __init__(
         self.drop_folds = drop_folds
         self.stat_name = stat_name or "te_stats"
 
-    @property
-    def req_stats(self):
-        stats = []
-        if self.target_mean is None:
-            stats.append(Moments(columns=self.cont_target))
-        stats.append(
-            GroupbyStatistics(
-                columns=self.cat_groups,
-                concat_groups=False,
-                cont_names=self.cont_target,
-                stats=["count", "sum"],
-                tree_width=self.tree_width,
-                out_path=self.out_path,
-                on_host=self.on_host,
-                stat_name=self.stat_name,
-                name_sep=self.name_sep,
-                fold_name="__fold__",
-                kfold=self.kfold,
-                fold_seed=self.fold_seed,
-                fold_groups=self.cat_groups,
-            )
-        )
-        return stats
+    # TODO: fit/fit_finalize methods
```

Comment:
Just a note:  Converting `op_logic` to `transform` will have one complication with the new API...  We were previously storing a `"__fold__"` column during the "fit," and then using that column during the "transform."  We can/should no longer do that.  The new API is not designed to track an underlying ddf in the workflow (for good reason), and it was already bad for performnace in the old API.

Perhaps a reasonable workaround is to use a hashing of the local row index and some random seed: `f(<seed>, <local-row-index>) % <nfolds>`?

Reply:
Thanks for the suggestion! I've just converted the TargetEncoding op in the last commit there - and definitely had some issues with storing the ```__fold__``` operator.

I like the idea about hashing the local row index, but went the route of moving the '_add_fold' function out of the fit - and then sharing with transform.  I had to tweak it a bit to make _add_fold function deterministic - it was originally going 
```python
cupy.random.set_seed(seed)
cupy.random.choice((cupy.arange(kfold, dtype=typ), len(s))
```
which had a race condition when called in the fit function (w/ ddf.index.map_partitions). We had multiple threads going through that and setting the seed, which caused the unittest to fail

I fixed by changing to 
```python
state = cupy.random.RandomState(fold_seed)
return state.choice(cupy.arange(kfold, dtype=typ), len(s))
``` 
which fixed the unittests (not relying on a global seed which could change).

I'm *hoping* this is sufficient - all the TE unittests pass at least =)

Reply:
>I'm hoping this is sufficient - all the TE unittests pass at least =)

Cool - For this PR, we just want things to "work."  I don't feel that the original implementation was very well tuned anyway :)